modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZonesUtil.java

```diff
@@ -676,15 +563,84 @@ public static List<CatalogTableDescriptor> findTablesByZoneId(int zoneId, Catalo
                 .collect(toList());
     }
 
-    /** Key prefix for zone's scale up change trigger key. */
-    @TestOnly
-    public static ByteArray zoneScaleUpChangeTriggerKeyPrefix() {
-        return new ByteArray(DISTRIBUTION_ZONE_SCALE_UP_CHANGE_TRIGGER_PREFIX);
+    public static Set<String> nodeNames(Set<NodeWithAttributes> nodes) {
+        return nodes.stream().map(NodeWithAttributes::nodeName).collect(toSet());
     }
 
-    /** Key prefix for zone's scale down change trigger key. */
-    @TestOnly
-    public static ByteArray zoneScaleDownChangeTriggerKeyPrefix() {
-        return new ByteArray(DISTRIBUTION_ZONE_SCALE_DOWN_CHANGE_TRIGGER_PREFIX);
+    /**
+     * Class representing data nodes' related values in Meta Storage.
+     */
+    public static class DataNodesHistoryContext {
```

Comment:
As usual, I would advice to extract this class into a separate file. Inner classes are neat when the enclosing class is small and it is easy to look at everything together. Here it is no longer the case.